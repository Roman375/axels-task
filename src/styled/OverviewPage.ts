import styled from 'styled-components'

export const Container = styled.section `
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  width: 100%;
  padding: 20px 40px;
  a {
    text-decoration: none;
    background-color: #AC1D00;
    color: #ffffff;
    padding: 10px;
    border-radius: 5px;
  }
  a:hover {
    background-color: black
  }
`

export const Text = styled.h1 `
  display: flex;
  justify-content: center;
`
