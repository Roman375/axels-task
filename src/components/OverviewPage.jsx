import React, { useEffect } from 'react'

import { Container, Text } from './styled'

const OverviewPage = ({ overview }) => {
  useEffect(() => {}, [overview])
  // const info = Object.getOwnPropertyDescriptor(overview[1], 'direct-subordinates').value
  // console.log(info)
  return (
    <Container>
      <Text>Overview</Text>
      <h1>{overview[0]}</h1>
      {overview.map((description, index) => (
        <div key={index}>
          <h3>{Object.values(description)}</h3>
        </div>
      ))}
    </Container>
  )
}

export default OverviewPage
