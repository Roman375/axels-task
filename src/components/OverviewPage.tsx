import React, { FC, useEffect } from 'react'

import { Container, Text } from '../styled/OverviewPage'

type Props = {
  overview: Array<object>
}

const OverviewPage: FC<Props> = ({overview}) => {
  
  useEffect(() => {
    console.log(overview);
  }, [overview])

  return (
    <Container>
      <Text>Overview</Text>
      {overview.map((description: object, index: number) => (
        <div key={index}>
          <h3>{Object.values(description)}</h3>
        </div>
      ))}
    </Container>
  )
}

export default OverviewPage
