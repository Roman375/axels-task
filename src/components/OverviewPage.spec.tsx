import { mount } from 'enzyme'

import OverviewPage from './OverviewPage'

describe('<OverviewPage />', () => {
  const props = {
    overview: [
      'John Hartman',
      { 'direct-subordinates': ['Samad Pitt', 'Leanna Hogg'] },
    ],
  }
  const app = mount(<OverviewPage {...props} />)

  it('render properly', () => {
    expect(app).toMatchSnapshot()
  })

  it('renders overview twice', () => {
    expect(app.find('div')).toHaveLength(2)
  })
})
