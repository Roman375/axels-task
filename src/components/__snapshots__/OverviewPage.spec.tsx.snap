// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OverviewPage /> render properly 1`] = `ReactWrapper {}`;
